jOOL-java-8/src/main/java/org/jooq/lambda/Agg.java

```diff
@@ -979,4 +979,694 @@ else if (percentile == 1.0)
             s -> s.map(Objects::toString).orElse("")
         );
     }
+
+    //CS304 Issue link: https://github.com/jOOQ/jOOL/issues/151
+
+    /**
+     * Calculate the linear regression function Slope of the given object collectors,
+     * based on the mapping function from object to double number.
+     *
+     * <p><pre> Usage of aggregation function regrSlopeBy():
+     * The mapping functions functionX and functionY are functions mapping the objects to double values X and Y, for instance:
+     * {@code
+     * FunctionX<Tuple2<Integer,Integer>, Double> mapping = e -> Double.valueOf(e.v1);
+     * FunctionY<Tuple2<Integer,Integer>, Double> mapping = e -> Double.valueOf(e.v2);
+     * }
+     * The specific usage of varianceBy is like:
+     * {@code
+     * Seq.of(new Tuple2<Integer,Integer>(1,1),new Tuple2<Integer,Integer>(2,2)).collect(Agg.regrSlopeBy(functionX,functionY));
+     * }
+     * Besides, self defined class is also allowed with mapping function:
+     * {@code
+     * Seq.of(new Node(-1,1),new Node(1,1)).collect(Agg.regrSlopeBy(functionX,functionY));
+     * }
+     * </pre>
+     *
+     * @param functionX mapping function from object to double value representing x
+     * @param functionY mapping function from object to double value representing y
+     * @return the Slope value
+     * @version 1.0
+     * @author Jichen Lu
+     * @date 2021-05-17
+     */
+    public static <T> Collector<T, ?, Optional<Double>> regrSlopeBy(Function<? super T, ? extends Number> functionX, Function<? super T, ? extends Number> functionY) {
```

Comment:
Again, `by()` has a different meaning in jOOL. Look at `Agg.minBy(Function<T, U>)`. It finds the value `T` for which the function argument produces the minimum value `U`. It does not produce the minimum value `U`.

Reply:
Just like with the previous review of https://github.com/jOOQ/jOOL/pull/387, we don't want do force users to use `Double` only. We should be able to calculate linear regressions for all numeric data types, including `Float`, `Double`, `BigDecimal` (granted, statistical functions make a bit less sense for integer types, but I don't see why we should offer `Double` only)

Reply:
So, we'll ignore other data types at first and use primitive doubles for now. Each one of these collectors should have 2 versions:

```java
public static Collector<Double, ?, Optional<Double>> regrSlopeDouble() {
    return regrSlopeDouble(t -> t);
}

public static <T> Collector<T, ?, Optional<Double>> regrSlopeDouble(ToDoubleFunction<? super T> functionX, ToDoubleFunction<? super T> functionY) {
}
```